src/pages/Join/JoinForm.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import './JoinForm.scss';
+
+const JoinForm = () => {
+  const navigate = useNavigate();
+
+  const postUserData = e => {
+    e.preventDefault();
+    navigate('/login');
+  };
+
+  const [inputValue, setInputValue] = useState({
+    userId: '',
+    userPw: '',
+    pwCheck: '',
+    userName: '',
+    userPhoneNumber: '',
+  });
+
+  const { userId, userPw, pwCheck, userName, userPhoneNumber } = inputValue;
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValue({ ...inputValue, [name]: value });
+  };
+
+  const [disabled, setDisabled] = useState(true);
+  const isDisabled = () => {
+    userId.includes('.') &&
+    userId.length >= 5 &&
+    userPw.length >= 5 &&
+    pwCheck === userPw &&
+    userName.length > 3 &&
+    userPhoneNumber.length > 0
+      ? setDisabled(false)
+      : setDisabled(true);
+  };
+
+  return (
+    <form
+      className="join-form"
+      onSubmit={postUserData}
+      onKeyUp={isDisabled}
+      onChange={handleInput}
+    >
+      <h2 className="join-title">회원가입</h2>
+      <div className="join-contnet">
+        <ul className="join-form-list">
+          {JOIN_INPUT_DATA.map(input => {
+            return (
+              <li key={input.id}>
+                <div className="input-box-error">
+                  <input
+                    className="inputtype"
+                    onChange={handleInput}
+                    name={input.name}
+                    type={input.type}
+                    placeholder={input.placeholder}
+                    autoFocus={input.autoFocus}
+                  />
+                  <p className="text-valid">{input.valid}</p>
+                </div>
+              </li>
+            );
+          })}
+          <li>
+            <div className="input-box-birthday">
+              <input type="text" placeholder="생일정보" />
+              <p className="text-valid">
+                &#183; 생일을 정확하게 입력해주세요. (예: 2000-12-31)
+              </p>
+              <p className="text-valid">
+                &#183; 만 14세 이상만 회원가입이 가능합니다.
+              </p>
+            </div>
+          </li>
+        </ul>
+      </div>
+
+      <div className="join-button">
+        <button className="join-button-submit" disabled={disabled}>
```

Comment:
```suggestion
        <button className="join-button-submit" disabled={isValid ? false : true}>
```
- 변수를 할당하시면 이렇게 동적으로 관리하실 수 있겠네요.

Reply:
Boolean 타입 변수 할당을 통해 동적으로 관리하는 방법을 알게 되었습니다. 감사합니다. 수정완료